### PR TITLE
BLUE-277: Restaking not working

### PR DESCRIPTION
**Pull request by Glitch18**
https://linear.app/shm/issue/BLUE-277/bug-cannot-restake-a-node

Restaking was broken. A nominator who has staked their node already was not able to add stake to their node again. 
This bug was due to an incorrect check in the preApply step for staking transactions. Fix added.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -4031,9 +4031,10 @@ const shardusSetup = (): void => {
           throw new Error('Stake transaction timestamp is too old')
         }
 
-        if (operatorEVMAccount.operatorAccountInfo?.nominee?.length > 0 ) {
-          throw new Error(`This node is already staked by another account!`)
-        }
+        // No need for this check. We already verify things in verifyStakeTx() above
+        // if (operatorEVMAccount.operatorAccountInfo?.nominee?.length > 0 ) {
+        //   throw new Error(`This node is already staked by another account!`)
+        // }
```
Comment (mssabr01) [inline src/index.ts]: Just remove the code, no need to just comment it out because the working functionality is alive an dwell in verifyStakeTx()

**Review by mssabr01 — CHANGES_REQUESTED** [review]
In addition to the minor comment, Line 3884 sets verifyResult to true. The try/catch block immediately following that statement sets verify resultResult to whatever is the output of verifyStakeTx() or verifyUnstakeTx(). Its possible that InternalTXType is neither Stake nor Unstake and verifyResult doesn't get touched at all.

I think the idea here is that if the internalTXType is something silly, then we should return a failure. For this to happen I think verifyResult needs to be initialzied to null on line 3884.

File: src/index.ts (hunk 1 above)
Comment (Glitch18) [reply src/index.ts]: Initializing verifyResult to false will make every TX apart from stake/unstake fail. This part of the code runs for all transactions. But we only want to run the verification for stake and unstake transactions. Hence verify stake needs to be true by default, and only get set to false if verifyStake/verifyUnstake fails.

I've removed the commented out code.

Comment (mssabr01) [reply src/index.ts]: Ah I see, good challenge

**Review by mssabr01 — DISMISSED** [review]

**Review by mhanson-github — APPROVED** [review]
Still approved, I rebased it on top of dev.

**Pull request merged**